PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/AdlsGen2VirtualFile.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.hdinsight.spark.ui.filesystem;
+
+import com.intellij.compiler.ant.taskdefs.Path;
+import com.intellij.mock.MockVirtualFile;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileSystem;
+import com.intellij.openapi.vfs.impl.http.HttpFileSystemImpl;
+import com.microsoft.azure.hdinsight.sdk.storage.adlsgen2.ADLSGen2FSOperation;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class AdlsGen2VirtualFile extends AzureStorageVirtualFile {
+    private String name;
+    private String path;
+    private VirtualFileSystem fileSystem;
+    private boolean isDirectory;
+    private List<VirtualFile> children;
+    private VirtualFile parent;
+
+    public AdlsGen2VirtualFile(String path, boolean isDirectory, VirtualFileSystem fileSystem) {
+        this.name = path.substring(path.lastIndexOf("/") + 1);
+        this.path = path;
+        this.isDirectory = isDirectory;
+        this.fileSystem = fileSystem;
+    }
+
+    @NotNull
+    @Override
+    public String getName() {
+        return this.name;
+    }
+
+    @NotNull
+    @Override
+    public VirtualFileSystem getFileSystem() {
+        return this.fileSystem;
+    }
+
+    @NotNull
+    @Override
+    public String getPath() {
+        return this.path;
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return this.isDirectory;
+    }
+
+    @Override
+    @Nullable
+    public VirtualFile getParent() {
+        return this.parent;
+    }
+
+    @Override
+    public void setParent(VirtualFile parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public VirtualFile[] getChildren() {
+        if (this.children != null) {
+            return this.children.toArray(new AdlsGen2VirtualFile[0]);
+        }
+
+        this.children = new ArrayList<>();
+        if(!this.isDirectory) {
+            return this.children.toArray(new AdlsGen2VirtualFile[0]);
+        }
+
+        URI uri = URI.create(this.path);
+        ADLSGen2FileSystem gen2FileSystem = (ADLSGen2FileSystem) fileSystem;
+        ADLSGen2FSOperation op = new ADLSGen2FSOperation(gen2FileSystem.http);
```

Comment:
`ADLSGen2FileSystem` and `ADLSGen2FSOperation` can be combined into one, `op` can be a private field of `ADLSGen2FileSystem`